app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -33,6 +33,37 @@ cat << EOF > $SSL_CA_CERT
 <%= foreman_server_ca_cert %>
 EOF
 
+<% unless @repo.blank? -%>
+echo '#'
+echo '# Adding repository'
+echo '#'
+
+if [ -f /usr/bin/yum ]; then
+  cat << EOF > /etc/yum.repos.d/foreman_registration.repo
+[foreman_register]
+name=foreman_register
+baseurl=<%= @repo %>
+gpgcheck=1
+gpgkey=<%= @repo_gpg_key %>
+enabled=1
+EOF
+
+  echo "Building yum metadata cache, this may take a few minutes"
+  yum makecache
+
+elif [ -f /usr/bin/apt-get ]; then
+  cat << EOF > /etc/apt/sources.list.d/foreman_registration.list
+<%= @repo %>
+EOF
+
+  apt-get -y install ca-certificates gpg
```

Comment:
this and the line below should only be performed if the gpg key is specified I think